pkg/apis/serving/v1alpha1/servingruntime_types.go

```diff
@@ -210,3 +221,15 @@ func (srSpec *ServingRuntimeSpec) IsDisabled() bool {
 func (srSpec *ServingRuntimeSpec) IsMultiModelRuntime() bool {
 	return srSpec.MultiModel != nil && *srSpec.MultiModel
 }
+
+func (srSpec *ServingRuntimeSpec) IsProtocolVersionSupported(modelProtocolVersion constants.InferenceServiceProtocol) bool {
+	if len(modelProtocolVersion) == 0 || srSpec.ProtocolVersions == nil || len(srSpec.ProtocolVersions) == 0 {
+		return true
+	}
+	for _, srProtocolVersion := range srSpec.ProtocolVersions {
+		if srProtocolVersion == modelProtocolVersion {
+			return true
+		}
+	}
+	return false
```

Comment:
Do we need to add a protocol field on inference service spec as it will never match with `grpc-v1` or `grpc-v2`?

Reply:
@yuzisun I added in 

> Can you also run `make generate` for the Open API generation?

Sure.